### PR TITLE
feat: added text support to system tray for macos, closes #65

### LINKED ISSUE #65: Implement ability to show text in the system tray/menu bar
Use case: apps like [MeetingBar](https://github.com/leits/MeetingBar) show text instead of a system tray icon.

image

I'd love the ability to do this with Tauri:

- Show text instead of the system tray icon, and be able to update it during runtime.
- Show text _next_ to the system tray icon.

However, I'm not sure if this is even possible in other operating systems. With that in mind, maybe it doesn't fit Tauri's philosophy to add a feature that will only work on macos!

### PR DESCRIPTION
**Pull request by vojty**
### What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [x] No

### Checklist
- [x] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [x] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [x] I have added a convincing reason for adding this feature, if necessary

### Other information

This is just my attempt to finish the work introduced in https://github.com/tauri-apps/tao/pull/369, all credits go to @BenJeau

Tested via `RUSTFLAGS='--cfg target_os="macos" --cfg feature="tray"' cargo run --example system_tray`

https://user-images.githubusercontent.com/660716/189293815-dc36ff51-3abb-4e34-a66b-aeebd9eb2f36.mov

### CODE REVIEW

File: examples/system_tray.rs [hunk 1]
```diff
@@ -31,11 +34,12 @@ fn main() {
   let second_tray_id = TrayId::new("2nd-tray");
   let icon = load_icon(std::path::Path::new(path));
   let mut tray_menu = Menu::new();
+  let menu_item = tray_menu.add_item(MenuItemAttributes::new("Item 1"));
```
Comment (amrbashir) [inline examples/system_tray.rs]:
```suggestion
  let menu_item = tray_menu.add_item(MenuItemAttributes::new("Set tray title (macos)"));
```

File: examples/system_tray.rs [hunk 2]
```diff
@@ -31,11 +34,12 @@ fn main() {
...
 
   #[cfg(target_os = "macos")]
   {
     tray_menu
-      .add_item(MenuItemAttributes::new("Item 1"))
+      .add_item(MenuItemAttributes::new("Item 2"))
```
Comment (amrbashir) [inline examples/system_tray.rs]:
```suggestion
      .add_item(MenuItemAttributes::new("Menu Item with icon"))
```

File: src/platform/macos.rs [hunk 3]
```diff
@@ -515,6 +515,9 @@ pub trait SystemTrayBuilderExtMacOS {
 
   /// Enables or disables showing the tray menu on left click, default is true.
   fn with_menu_on_left_click(self, enable: bool) -> Self;
+
+  /// Sets the tray menu title
```
Comment (amrbashir) [inline src/platform/macos.rs]:
```suggestion
  /// Sets the tray icon title
```

File: src/platform/macos.rs [hunk 4]
```diff
@@ -540,6 +548,9 @@ pub trait SystemTrayExtMacOS {
 
   /// Enables or disables showing the tray menu on left click, default is true.
   fn enable_menu_on_left_click(&mut self, enable: bool);
+
+  /// Sets the tray menu title
```
Comment (amrbashir) [inline src/platform/macos.rs]:
```suggestion
  /// Sets the tray icon title
```

File: src/platform_impl/macos/system_tray.rs [hunk 5]
```diff
@@ -171,6 +187,7 @@ impl SystemTray {
 
       button.setImage_(nsimage);
       let _: () = msg_send![nsimage, setSize: new_size];
+      let _: () = msg_send![button, setImagePosition: 2]; // https://developer.apple.com/documentation/appkit/nscellimageposition/nsimageleft
```
Comment (amrbashir) [inline src/platform_impl/macos/system_tray.rs]: extract this value into a const variable and add the URL as a doc comment

**Review by amrbashir — CHANGES_REQUESTED** [review]
Thank you for taking this feature to the finish line, just a few nit-picks to go.

**Review by amrbashir — APPROVED** [review]

**Comment by BenJeau** [comment]
Thanks @vojty for updating my initial PR and having it merged :)

**Comment by vojty** [comment]
@BenJeau np, you did the hard work anyway :) However, I didn't realize that we need to expose this new API to `tauri` as well. Any help here https://github.com/tauri-apps/tauri/pull/5182 would be appreciated.

**Pull request merged**